docs/frc/1-Basics/9-Drivetrain.md

```diff
@@ -1,3 +1,300 @@
+
+We'll declare our state(s) in the form of an [enum:](https://frc1257.github.io/robotics-training/#/java/2-Control_Flow/7-Enums)
+
+```java
+    public enum State {
+        MANUAL
+    }
+    private final State defaultState = State.MANUAL;
+    private State state = defaultState;
+```
+
+We then add a private variable that'll hold the default state of the subsystem, which, in this case, will be `MANUAL`. Having a separate default state variable is good for expandability in the program and also reduces confusion during robot testing or competition. Finally, we initialize our subsystem-wide state variable with the default state.
+
+### Constructor
+
+```java
+    public Drivetrain() {
+        frontLeftMotor = new CANSparkMax(DRIVE_FRONT_LEFT, MotorType.kBrushless);
+        frontRightMotor = new CANSparkMax(DRIVE_FRONT_RIGHT, MotorType.kBrushless);
+        backLeftMotor = new CANSparkMax(DRIVE_BACK_LEFT, MotorType.kBrushless);
+        backRightMotor = new CANSparkMax(DRIVE_BACK_RIGHT, MotorType.kBrushless);
+
+        backLeftMotor.follow(frontLeftMotor);
+        backRightMotor.follow(frontRightMotor);
+
+        frontLeftMotor.restoreFactoryDefaults();
+        frontRightMotor.restoreFactoryDefaults();
+        backLeftMotor.restoreFactoryDefaults();
+        backRightMotor.restoreFactoryDefaults();
+
+        frontLeftMotor.setSmartCurrentLimit(NEO_CURRENT_LIMIT);
+        frontRightMotor.setSmartCurrentLimit(NEO_CURRENT_LIMIT);
+        backLeftMotor.setSmartCurrentLimit(NEO_CURRENT_LIMIT);
+        backRightMotor.setSmartCurrentLimit(NEO_CURRENT_LIMIT);
+
+        frontLeftMotor.setIdleMode(IdleMode.kBrake);
+        frontRightMotor.setIdleMode(IdleMode.kBrake);
+        backLeftMotor.setIdleMode(IdleMode.kCoast);
+        backRightMotor.setIdleMode(IdleMode.kCoast);
+
+        drivetrain = new DifferentialDrive(frontLeftMotor, frontRightMotor);
+
+        speedForward = 0;
+        speedTurn = 0;
+    }
+```
+
+In the constructor, we'll take all our declarations and initialize them. Most of the work here is done for the motors:
+
+1. Initialize each motor with respective ID variables, and set the brushed/brushless mode with `MotorType.kBrushless`.
+2. Use `restoreFactoryDefaults()` to reset any of the motors' internal settings.
+3. Use `setSmartCurrentLimit()` to apply the appropriate current limit value (for NEOs, we generally do 80 amps)
+4. Set the front two motors to brake mode and back two motors to coast mode with `setIdleMode()`.
+5. Initialize the `DifferentialDrive` object by passing in the two front motors (front left and then front right), and then set driving speeds to 0. 
+
+Note that the drivetrain motor IDs and NEO current limit are all variables imported from `Constants.java`, so we would add the following import statements at the top of our file:
```

Comment:
I stressed in a previous section that we don't really need to worry about import statements, so you can probably just remove this section